File: 380-insert-delete-getrandom-o1/380-insert-delete-getrandom-o1.java
class RandomizedSet {
    HashSet<Integer> rando;
    public RandomizedSet() {
        rando = new HashSet<>();
    }
    
    public boolean insert(int val) {
        return rando.add(val);
        
    }
    
    public boolean remove(int val) {
        return rando.remove(val);
    }
    
    public int getRandom() {
        int r = (int)(Math.random() * rando.size());
        Iterator<Integer> iter = rando.iterator();
        int res = iter.next();
        for(int i = 0; i < r; i++){
            res = iter.next();
        }
        return res;
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: 380-insert-delete-getrandom-o1/380-insert-delete-getrandom-o1.py
class RandomizedSet:

    def __init__(self):
        self.randSet = set()

    def insert(self, val: int) -> bool:
        if val in self.randSet:
            return False
        self.randSet.add(val)
        return True

    def remove(self, val: int) -> bool:
        if val in self.randSet:
            self.randSet.remove(val)
            return True
        else:
            return False

    def getRandom(self) -> int:
        rand = randint(0,len(self.randSet)-1)
        lst = list(self.randSet)
        return lst[rand]


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()